pkg/mpc-examples/contracts/01-weight-changer/WeightChangerController.sol

```diff
@@ -0,0 +1,146 @@
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+// This contract shows an example of how a managed pool controller can modify a pools weight
+// Gradual weight updates over 7 days for each change
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IManagedPool.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
+import "@balancer-labs/v2-pool-utils/contracts/lib/ComposablePoolLib.sol";
+import "../interfaces/IManagedPoolFactory.sol";
+
+// solhint-disable not-rely-on-time
+
+contract WeightChangerController {
+    using FixedPoint for uint256;
+    IERC20[] private _tokens;
+
+    uint256 private constant _REWEIGHT_DURATION = 7 days;
+
+    // Minimum and maximum weight limits
+    uint256 private constant _MIN_WEIGHT = 1e16; // 1%
+    uint256 private constant _MAX_WEIGHT = 99e16; // 99%
+
+    IVault private immutable _vault;
+    bytes32 private immutable _poolId;
+
+    constructor(IVault vault) {
+        // Get poolId from the factory
+        bytes32 poolId = IManagedPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()).getPoolId();
+
+        // Verify that this is a real Vault and the pool is registered - this call will revert if not.
+        vault.getPool(poolId);
+
+        // Set the global tokens variables
+        (IERC20[] memory tokens, , ) = vault.getPoolTokens(poolId);
+        _setTokens(tokens);
+
+        _vault = vault;
+        _poolId = poolId;
+    }
+
+    function make5050() public {
+        uint256[] memory newWeights = new uint256[](2);
+        newWeights[0] = 50e16;
+        newWeights[1] = 50e16;
+        _updateWeights(block.timestamp, block.timestamp + _REWEIGHT_DURATION, getTokens(), newWeights);
+    }
+
+    function make8020() public {
+        uint256[] memory newWeights = new uint256[](2);
+        newWeights[0] = 80e16;
+        newWeights[1] = 20e16;
+        _updateWeights(block.timestamp, block.timestamp + _REWEIGHT_DURATION, getTokens(), newWeights);
+    }
+
+    function make9901() public {
+        uint256[] memory newWeights = new uint256[](2);
+        newWeights[0] = 99e16;
+        newWeights[1] = 1e16;
+        _updateWeights(block.timestamp, block.timestamp + _REWEIGHT_DURATION, getTokens(), newWeights);
+    }
```

Comment:
Can you add natspec comments at least for these three fns? Most of the other functions should be pretty obvious, but explanations here might help readers